pkg/controller/networkpolicyrecommendation/controller.go

```diff
@@ -598,7 +597,7 @@ func (c *NPRecommendationController) startSparkApplication(npReco *crdv1alpha1.N
 	}
 	sparkResourceArgs.executorMemory = npReco.Spec.ExecutorMemory
 
-	recommendationID := uuid.New().String()
+	recommendationID := npReco.Name[3:]
```

Comment:
Maybe you want to check it is valid npReco.Name?
Like,
`err = commands.ParseRecommendationName(npReco.Name)
	if err != nil {
		return IlleagelArguementError{fmt.Errorf("invalid request: Job name is not a valid policy recommendation job name")}
	}`